### PR TITLE
Add mm_fp4 trtllm backend

### PR DESCRIPTION
**Pull request by wenscarl**
## Motivation
Add trtllm backend for mm_fp4. env var `SGLANG_FLASHINFER_FP4_GEMM_BACKEND` is added which defaults to cudnn to align with flashinfer design of [API](https://docs.flashinfer.ai/generated/flashinfer.gemm.mm_fp4.html).

There is still a minor issue with the behavior of default backend. Hold on until flashinfer fixes it by https://github.com/flashinfer-ai/flashinfer/pull/2026/files.

## Accuracy Tests
server:
```
SGLANG_FLASHINFER_FP4_GEMM_BACKEND="trtllm" \
python3 -m sglang.launch_server \
  --model-path nvidia/DeepSeek-R1-0528-FP4 \
  --trust-remote-code \
  --disable-radix-cache \
  --max-running-requests 256 \
  --chunked-prefill-size 2048 \
  --mem-fraction-static 0.89 \
  --disable-cuda-graph \
  --max-prefill-tokens 16384 \
  --tp 8 \
  --dp 8 \
  --enable-dp-attention \
  --ep-size 8
```
client:
```
python3 benchmark/gsm8k/bench_sglang.py   --num-questions 256   --parallel 32   --num-shots 8

Accuracy: 0.977
Invalid: 0.000
Output throughput: 192.367 token/s
```

SGLANG_FLASHINFER_FP4_GEMM_BACKEND="cutlass"
```
Accuracy: 0.973
Invalid: 0.000
Output throughput: 135.601 token/s
```

SGLANG_FLASHINFER_FP4_GEMM_BACKEND="cudnn"
Accuracy: 0.980
Invalid: 0.000
Output throughput: 75.348 token/s
## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

File: python/sglang/srt/layers/quantization/modelopt_quant.py [hunk 1]
```diff
@@ -84,6 +84,9 @@
 USE_CUTLASS_BACKEND_FOR_FP4_GEMM = get_bool_env_var(
     "SGLANG_USE_CUTLASS_BACKEND_FOR_FP4_GEMM", "true"
 )
+USE_TRTLLM_BACKEND_FOR_FP4_GEMM = get_bool_env_var(
```
Comment (Fridge003) [inline python/sglang/srt/layers/quantization/modelopt_quant.py]: Can we expose them as server arguments?

Comment (Fridge003) [reply python/sglang/srt/layers/quantization/modelopt_quant.py]: A string environment variable stored at `environ.py` should also work

File: python/sglang/srt/layers/quantization/modelopt_quant.py [hunk 2]
```diff
@@ -88,11 +93,16 @@
...
```
Comment (Fridge003) [inline python/sglang/srt/layers/quantization/modelopt_quant.py]: These two environments variable can be removed?

File: python/sglang/srt/layers/quantization/modelopt_quant.py [hunk 3]
```diff
@@ -1037,14 +1065,20 @@ def apply(
         if enable_flashinfer_fp4_gemm:
             w = layer.weight.T
             w_scale_interleaved = layer.weight_scale_interleaved.T
+
+        assert not (USE_CUTLASS_BACKEND_FOR_FP4_GEMM and USE_TRTLLM_BACKEND_FOR_FP4_GEMM), (
```
Comment (Fridge003) [inline python/sglang/srt/layers/quantization/modelopt_quant.py]: This line can be removed

File: python/sglang/srt/utils/common.py [hunk 4]
```diff
@@ -265,6 +265,14 @@ def get_int_env_var(name: str, default: int = 0) -> int:
         return default
 
 
+def get_str_env_var(name: str, default: str = "") -> str:
```
Comment (Fridge003) [inline python/sglang/srt/utils/common.py]: We don't need this util function.
String environs can be obtained by
```bash
from sglang.srt.environs import envs
FLASHINFER_FP4_GEMM_BACKEND = envs.SGLANG_FLASHINFER_FP4_GEMM_BACKEND.get()
```

**Review by Fridge003 — APPROVED** [review]

**Comment by Fridge003** [comment]
@wenscarl Sorry for the typo, it should be 
```bash
from sglang.srt.environ import envs
```

File: python/sglang/srt/layers/quantization/modelopt_quant.py [hunk 5]
```diff
@@ -8,6 +8,7 @@
 from torch.nn.parameter import Parameter
 
 from sglang.srt.distributed import get_tp_group
+from sglang.srt.environs import envs
```
Comment (Fridge003) [inline python/sglang/srt/layers/quantization/modelopt_quant.py]: Should be `from sglang.srt.environ import envs`

**Comment by Fridge003** [comment]
@wenscarl Please fix lint, thanks
